src/ForegroundEffects.js

```diff
@@ -0,0 +1,84 @@
+const constants = require('./constants');
+const utils = require('./utils');
+
+const rain = require('./effects/foreground/rain');
+const rainingTacos = require('./effects/foreground/rainingTacos');
+const pineapples = require('./effects/foreground/pineapples');
+const spotlight = require('./effects/foreground/spotlight');
+const colorLights = require('./effects/foreground/colorLights');
+const smilingPoop = require('./effects/foreground/smilingPoop');
+const heartsRed = require('./effects/foreground/heartsRed');
+const heartsColorful = require('./effects/foreground/heartsColorful');
+const floatingRainbows = require('./effects/foreground/floatingRainbows');
+const bubbles = require('./effects/foreground/bubbles');
+const explodingStars = require('./effects/foreground/explodingStars');
+const pizzas = require('./effects/foreground/pizzas');
+const smileFace = require('./effects/foreground/smileFace');
+const confetti = require('./effects/foreground/confetti');
+const musicNotes = require('./effects/foreground/musicNotes');
+const paintDrip = require('./effects/foreground/paintDrip');
+const emojis = require('./effects/foreground/emojis');
+
+module.exports = class ForegroundEffects {
+  constructor(p5, alpha, getEffectsInPreviewMode) {
+    this.p5_ = p5;
+
+    // Duplicated in BackgroundEffects
```

Comment:
I ended up duplicating these helpers (rather than moving them into a utils file) because:
- p5.random is easier to stub in tests than Math.random, as it allows you to provide a random seed. Trying to stub Math.random using sinon ended up breaking some tests in ways I couldn't understand (JS out of memory issues?)
- some of the helpers (eg, `colorFromHue`) are defined with a default parameter that is passed in via the constructor (`alpha`), which made it a little hard to follow to define `colorFromHue` in a helper (ie, we needed a helper that would return a function). Seemed clearer just to duplicate.